pkg/controller/node/node_controller.go

```diff
@@ -1042,3 +1065,108 @@ func getErrorString(err error) string {
 	}
 	return ""
 }
+
+// setPendingFileMetrics checks to see if there are any important files in the
+// machineconfig that the pool should be moving to, and sets metrics if there are
+func (ctrl *Controller) setPendingFileMetrics(pool *mcfgv1.MachineConfigPool) {
+	// Retrieve and parse the pool's machine config
+	currentConfig, pendingConfig, err := ctrl.parseConvertMachineConfigFilesForPool(pool)
+	if err != nil {
+		glog.Warningf("Error converting pool configs for %s pool: %v", pool.Name, err)
+		return
+	}
+
+	// Figure out what files differ between pool.Spec and pool.Status
+	fileDiff := ctrlcommon.CalculateConfigFileDiffs(currentConfig, pendingConfig)
+
+	// Go through our files until we hit the kubelet CA bundle
+	for _, path := range fileDiff {
+		// We only care about the kubelet CA bundle
+		if path != kubeletCAFilePath {
+			continue
+		}
+
+		// If it's there, get the *newest* (in case there have been multiple rotations) kube-apiserver-to-kubelet signer certifiate out of the bundle
+		newestSignerCertificate, err := ctrl.getNewestAPIToKubeletSignerCertificate(currentConfig)
+		if err != nil {
+			glog.Warningf("Error retrieving kubelet-ca certificates from pool %s: %v", pool.Name, err)
+		} else {
+			// Set the metric value to the UTC expiry date of that cert so we can count down to it
+			glog.V(2).Infof("Kubelet CA is stuck in paused pool %s. Setting metric to expiry date of %s (%s)", pool.Name, newestSignerCertificate.Subject.CommonName, newestSignerCertificate.NotAfter.UTC())
+			ctrlcommon.MachineConfigControllerPausedPoolKubeletCA.WithLabelValues(pool.Name).Set(float64(newestSignerCertificate.NotAfter.UTC().Unix()))
+		}
+		break
+	}
+}
+
+// resetPendingFileMetrics turns off any "paused file" metrics that were firing for the pool
+func (ctrl *Controller) resetPendingFileMetrics(pool *mcfgv1.MachineConfigPool) {
+	// Set the metric for this pool back to zero
+	ctrlcommon.MachineConfigControllerPausedPoolKubeletCA.WithLabelValues(pool.Name).Set(0)
+}
+
+// parseConvertMachineConfigFilesForPool retrieves the current and pending configurations for a pool, parses and converts them, and
```

Comment:
can we add some mention about metrics in docstring, bc it wasn't clear to me how this was going to be used given the name and current docstring.